src/lib/VectorTileSource.svelte

```diff
@@ -0,0 +1,58 @@
+<script lang="ts">
+  import { onDestroy, tick } from 'svelte';
+  import { getId, updatedSourceContext } from './';
+  import { Protocol } from 'pmtiles';
+  import maplibregl from 'maplibre-gl';
+  import flush from 'just-flush';
+
+  export let id: string = getId('vector');
+  export let url: string | null = null;
+  export let tiles: Array<string> | null = null;
+
+  if (url && url.includes('pmtiles://')) {
+    if (!maplibregl.config.REGISTERED_PROTOCOLS.hasOwnProperty('pmtiles')) {
+      let protocol = new Protocol();
+      maplibregl.addProtocol('pmtiles', protocol.tile);
+    }
+  }
+
+  /** Generate a unique id for each feature. This will overwrite existing IDs. */
```

Comment:
Stray comment? I don't think maplibre can do generateId for anything besides GJ sources

Reply:
Yup. Removed now